lib/measures/tbd/measure.rb

```diff
@@ -357,12 +367,29 @@ def run(mdl, runner, args)
       end
     end
 
-    # Pre-validate ground-facing constructions for KIVA.
-    if argh[:kiva_force] || argh[:gen_kiva]
-      kva = true
+    kva = false
+    kva = true unless mdl.getSurfacePropertyExposedFoundationPerimeters.empty?
+    kva = true unless mdl.getFoundationKivas.empty?
+
+    # Purge existing KIVA objects from model.
+    if argh[:reset_kiva]
+      if kva
+        if argh[:gen_kiva]
+          resetKIVA(mdl, "Foundation")
+        else
+          resetKIVA(mdl, "Ground")
```

Comment:
Purge existing KIVA objects (either in isolation from other TBD options, or prior to having TBD auto-generate KIVA objects). Once purged, outside boundary condition are set to either "Foundation" or "Ground", depending if users want to _generate_ or _force-generate_ KIVA inputs via TBD. If neither, outside boundary condition is set to "Ground" by default. Applies only to surfaces previously linked to KIVA "Foundation" objects.